src/enhanced/enhancedApp.ts

```diff
@@ -1,15 +1,67 @@
 import { handleJob } from './utils/job';
 import { listenToJobQueue } from './utils/queue';
+import mongodb, { MongoClient } from 'mongodb';
+import c from 'config';
 
-async function app() {
-  console.log('starting application');
-  const { jobId } = await listenToJobQueue();
+let client: MongoClient | undefined;
 
-  await handleJob(jobId);
+async function connectToDb(): Promise<mongodb.Db> {
+  const atlasURL = `mongodb+srv://${c.get('dbUsername')}:${c.get('dbPassword')}@${c.get(
+    'dbHost'
+  )}/?retryWrites=true&w=majority`;
 
-  console.log('process completed');
+  console.log('[connectToDb]: Instantiating MongoDB client object');
+  client = new MongoClient(atlasURL);
+  console.log('[connectToDb]: Connecting to client');
+  await client.connect();
+  return client.db(c.get('dbName'));
+}
+
+async function cleanupJob(): Promise<never> {
+  try {
+    console.log('[cleanupJob]: Closing MongoDB client connection...');
+    await client?.close();
```

Comment:
nit: Maybe check that `client` is defined earlier and return if not, so that we never get a log saying we're closing the connection if there isn't one. I also understand this will probably never happen and is only there for typescript reasons.